app/controllers/reset_pins_controller.rb

```diff
@@ -3,59 +3,71 @@
 # Controller for reseting a Barcode+PIN user's PIN
 class ResetPinsController < ApplicationController
   before_action :logout_user!
+  rescue_from ActiveSupport::MessageEncryptor::InvalidMessage, with: :invalid_token
+  rescue_from FolioClient::IlsError, SymphonyClient::IlsError, with: :request_failed
 
-  # Renders the first step for reseting the PIN
+  # Renders the first step for resetting the PIN
   #
   # GET /reset_pin
   def index; end
 
-  # Trigger a reset request for the PIN in Symphony; this will
-  # send the patron an email (using their email adddress on file)
-  # with a link back to our change form and a token for completing the reset
+  # Trigger a reset request for the PIN; this will send the patron an email
+  # (using their email address on file) with a link back to our change form and
+  # a token for completing the reset
+  #
+  # Ignore errors indicating the patron wasn't found in the ILS; we don't want
+  # to leak information about presence/validity of library IDs
   #
   # POST /reset_pin
   def reset
-    @response = ils_client.reset_pin(
-      params['library_id'],
-      change_pin_with_token_unencoded_url
-    )
-    flash[:success] = t 'mylibrary.reset_pin.success_html', library_id: params['library_id']
-    render action: :index
+    suppress ActiveRecord::RecordNotFound do
+      ils_client.reset_pin(reset_pin_params, change_pin_with_token_unencoded_url)
+      flash[:success] = t 'mylibrary.reset_pin.success_html', library_id: params['library_id']
+      redirect_to login_path
+    end
```

Comment:
I don't think I understand what difference this makes. If we suppress the exception, won't we always render the success message anyway?

Reply:
Maybe I don't understand how `suppress` works, but from Marlo's report it sounds like we're not sending the flash + redirect, right?

Reply:
Boy, I should've just read the docs for `suppress` again:
```rb
suppress(ZeroDivisionError) do
  1/0
  puts 'This code is NOT reached'
end

puts 'This code gets executed and nothing related to ZeroDivisionError was seen'
```

...so yeah, this will do what we want 😂